media/jackson/src/main/java/io/helidon/media/jackson/JacksonSupport.java

```diff
@@ -33,94 +34,114 @@
+        return DEFAULT.get();
     }
 
     /**
-     * Creates new Jackson writer instance.
+     * Creates a new {@link JacksonSupport}.
      *
-     * @return Jackson writer instance
+     * @param objectMapper must not be {@code null}
+     * @return a new {@link JacksonSupport}
      */
-    public static JacksonBodyWriter writer() {
-        return create().newWriter();
+    public static JacksonSupport create(ObjectMapper objectMapper) {
+        Objects.requireNonNull(objectMapper);
+        return new JacksonSupport(objectMapper);
     }
 
     /**
-     * Creates new Jackson reader instance.
+     * Return a default Jackson entity reader.
      *
-     * @return Jackson reader instance
+     * @return default Jackson body writer instance
      */
-    public JacksonBodyReader newReader() {
-        return JacksonBodyReader.create(objectMapper);
+    public static MessageBodyReader<Object> reader() {
+        return DEFAULT.get().reader;
     }
 
     /**
-     * Creates new Jackson writer instance.
+     * Create a new Jackson entity reader based on {@link ObjectMapper} instance.
      *
-     * @return Jackson writer instance
+     * @param objectMapper object mapper instance
+     * @return new Jackson body reader instance
      */
-    public JacksonBodyWriter newWriter() {
-        return JacksonBodyWriter.create(objectMapper);
+    public static MessageBodyReader<Object> reader(ObjectMapper objectMapper) {
+        Objects.requireNonNull(objectMapper);
+        return JacksonBodyReader.create(objectMapper);
     }
 
-    @Override
-    public Collection<MessageBodyReader<?>> readers() {
-        return List.of(newReader());
+    /**
+     * Return a default Jackson entity writer.
+     *
+     * @return default Jackson body writer instance
+     */
+    public static MessageBodyWriter<Object> writer() {
+        return DEFAULT.get().writer;
     }
 
-    @Override
-    public Collection<MessageBodyWriter<?>> writers() {
-        return List.of(newWriter());
+    /**
+     * Create a new Jackson entity writer based on {@link ObjectMapper} instance.
+     *
+     * @param objectMapper object mapper instance
+     * @return new Jackson body writer instance
+     */
+    public static MessageBodyWriter<Object> writer(ObjectMapper objectMapper) {
+        Objects.requireNonNull(objectMapper);
+        return JacksonBodyWriter.create(objectMapper);
     }
 
     /**
-     * Creates a new {@link JacksonSupport}.
+     * Return Jackson reader instance.
      *
-     * @param objectMapper must not be {@code null}
-     * @return a new {@link JacksonSupport}
-     *
-     * @exception NullPointerException if {@code objectMapper}
-     * is {@code null}
+     * @return Jackson reader instance
      */
-    public static JacksonSupport create(ObjectMapper objectMapper) {
-        Objects.requireNonNull(objectMapper);
-        return new JacksonSupport(objectMapper);
+    public MessageBodyReader<Object> readerInstance() {
```

Comment:
Inconsistency:
`readerInstance()` against `newWriter()` methods

The `readerInstance()` method is worse, as it prescribes implementation - it is expected to return the same instance every time.

The `newWriter()` method clearly says that it may be a different instance each time (and yet if implementation wants, it can return the same one).

I would stay with the `newXXX` methods

Reply:
Actually this method is not described by any interface, so it is basically up to every support class how it chooses to name it or even decides to have them. Since we are returning every time the same instance of the reader/writer, I think `readerInstance()` is actually better name here.